src/sentry/integrations/utils/metrics.py

```diff
@@ -25,65 +24,86 @@ def __str__(self) -> str:
         return self.value.lower()
 
 
+@dataclass(frozen=True, eq=True)
+class MetricId:
+    key: str
+    tags: Mapping[str, str]
+
+
 class EventLifecycleMetric(ABC):
     """Information about an event to be measured.
 
-    This class is intended to be used across different integrations that share the
-    same business concern. Generally a subclass would represent one business concern
-    (such as MessagingInteractionEvent, which extends this class and is used in the
-    `slack`, `msteams`, and `discord` integration packages).
+    This is a generic base class not tied specifically to integrations. See
+    IntegrationEventLifecycleMetric for integration-specific key structure. (This
+    class could be moved from this module to a more generic package if we ever want
+    to use it outside of integrations.)
```

Comment:
FYI @corps, since IIRC you expressed some interest in doing this.

Reply:
Thank you!